src/plugins/discord/sqliteMirror.js

```diff
@@ -497,4 +497,21 @@ export class SqliteMirror {
       .all({channel_id: channel, message_id: message})
       .map((e) => Model.refToEmoji(e.emoji));
   }
+
+  // returns undefined if n is greater than the number of messages in the channel
+  nthMessageIdFromTail(channel: Model.Snowflake, n: number): ?Model.Snowflake {
+    return this._db
+      .prepare(
+        dedent`
+          SELECT id
+          FROM messages
+          WHERE channel_id = :channel_id
+          ORDER BY timestamp_ms DESC
+          LIMIT 1
+          OFFSET :offset
+        `
+      )
+      .pluck()
+      .get({channel_id: channel, offset: n - 1});
```

Comment:
Might be off about this, but this indexes the n passed in from 1. I was confusing myself in the tests: I was passing in n = 4 to `nthMessageFromTail`,  with a message array `[1, 2 ,3, 4, 5]` , and getting back 1. This would return 2, which seems more intuitive. But happy to revert this as well.